src/rpc/ngHandlers/Ledger.cpp

```diff
@@ -0,0 +1,170 @@
+        output.header[JS(closed)] = true;
+        output.header[JS(hash)] = ripple::strHex(lgrInfo.hash);
+        output.header[JS(ledger_hash)] = ripple::strHex(lgrInfo.hash);
+        output.header[JS(ledger_index)] = std::to_string(lgrInfo.seq);
+        output.header[JS(parent_close_time)] = lgrInfo.parentCloseTime.time_since_epoch().count();
+        output.header[JS(parent_hash)] = ripple::strHex(lgrInfo.parentHash);
+        output.header[JS(seqNum)] = std::to_string(lgrInfo.seq);
+        output.header[JS(totalCoins)] = ripple::to_string(lgrInfo.drops);
+        output.header[JS(total_coins)] = ripple::to_string(lgrInfo.drops);
+        output.header[JS(transaction_hash)] = ripple::strHex(lgrInfo.txHash);
+    }
+    output.header[JS(closed)] = true;
+
+    if (input.transactions)
+    {
+        output.header[JS(transactions)] = boost::json::value(boost::json::array_kind);
+        boost::json::array& jsonTxs = output.header.at(JS(transactions)).as_array();
+        if (input.expand)
+        {
+            auto txns = sharedPtrBackend_->fetchAllTransactionsInLedger(lgrInfo.seq, ctx.yield);
+
+            std::transform(
+                std::move_iterator(txns.begin()),
+                std::move_iterator(txns.end()),
+                std::back_inserter(jsonTxs),
+                [&](auto obj) {
+                    boost::json::object entry;
+                    if (!input.binary)
+                    {
+                        auto [txn, meta] = RPC::toExpandedJson(obj);
+                        entry = std::move(txn);
+                        entry[JS(metaData)] = std::move(meta);
+                    }
+                    else
+                    {
+                        entry[JS(tx_blob)] = ripple::strHex(obj.transaction);
+                        entry[JS(meta)] = ripple::strHex(obj.metadata);
+                    }
+                    // entry[JS(ledger_index)] = obj.ledgerSequence;
+                    return entry;
+                });
+        }
+        else
+        {
+            auto hashes = sharedPtrBackend_->fetchAllTransactionHashesInLedger(lgrInfo.seq, ctx.yield);
+            std::transform(
+                std::move_iterator(hashes.begin()),
+                std::move_iterator(hashes.end()),
+                std::back_inserter(jsonTxs),
+                [](auto hash) { return boost::json::string(ripple::strHex(hash)); });
+        }
+    }
+
+    if (input.diff)
+    {
+        output.header["diff"] = boost::json::value(boost::json::array_kind);
+        boost::json::array& jsonDiff = output.header.at("diff").as_array();
+        auto diff = sharedPtrBackend_->fetchLedgerDiff(lgrInfo.seq, ctx.yield);
+        for (auto const& obj : diff)
+        {
+            boost::json::object entry;
+            entry["object_id"] = ripple::strHex(obj.key);
+            if (input.binary)
+                entry["object"] = ripple::strHex(obj.blob);
+            else if (obj.blob.size())
+            {
```

Comment:
I'd rather not mix curly braces and no braces in one if-else statement block. If one of the clauses have enough complexity - others should probably have the same style (i.e. with braces in this case).